### PR TITLE
Update Progress indicators

### LINKED ISSUE #1061: Re-sizeable progress indicator
Add support for re-sizeable progress indicators, both for smaller to be used inside components and larger for "page loading state". 

- [x] Verify which progress indicator we want to be re-sizeable.
  - Requested are inside components such as checkboxes (selection controls?), textfields(helper or meta text?), buttons 
  - Requested re-sizeable progress indicators are circular and dots ( Can others work? )~
- [x] Verify with design what we want to change, support and valid sizes? Same size-scale as Icon?
- [x] Implement the stuff. Make sure you conform with other components on api.
  - Loading states inside components may just need to be an example of components being swapped. 
  - Avoid bloated "loading" props if possible
- [x] Reward yourself with cake on a job well done 🍰

Notes from meeting with @lucasveil 
* We will introduce size property to circular and star progress indicators
* Size will follow same scaling as defined for `Icon`

resolves #417 
resolves #1041

### LINKED ISSUE #1089: Implement missing progress indicator colors
Dots and circular needs to be updated with new color options in react.

### PR DESCRIPTION
**Pull request by mimarz**
resolves #1061 
resolves #1089 

## Updates
- Conformed on `color` option for all progress indicators. 
- `Dots` and `Circular` have a `color` prop. 
- Cleaned up and updated code to "latest" iteration of component architecture and typescript knowledge
- Removed explicit `className` usage as it served no purpose, looked like old testing/approach.
- All progress indicators are now pure svg icons with css animations. ( Removed unneeded `div` wrapper on `Circular`)
- Changed `LinearProgress` to use ternary operator to avoid duplicate divs for `indeterminate` variant
- Added compound component `Progress` to be more inline with library. You can now choose between using `LinearProgress` or `Progress.Linear`
- Updated stories to be more consistent

## Breaking changes ⚠️
- Removed `variant` from `DotsProgress` as this did not conform with our component apis. Color of dots is now determined by `color` prop and one of the agreed upon color names, `primary`, `neutral` or `tetriary`.
- Removed forced svg size inside `Button` because `CircularProgress` needs to be `size=16` inside `Button`

### CODE REVIEW

File: libraries/core-react/src/components/Progress/Circular/CircularProgress.tokens.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  background: primary13,
+  entities: {
+    progress: {
+      background: primary100,
+    },
+  },
+}
+
+export const neutral: CircularProgressToken = {
+  // TODO replace with token color once Figma is updated 29.01.2021
```
Comment (wenche) [inline libraries/core-react/src/components/Progress/Circular/CircularProgress.tokens.ts:30]: Create an issue for this?

File: libraries/core-react/src/components/Progress/Dots/DotProgress.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as React from 'react'
-import { forwardRef, Ref, SVGProps } from 'react'
+import { forwardRef, SVGProps, Ref } from 'react'
```
Comment (wenche) [inline libraries/core-react/src/components/Progress/Dots/DotProgress.tsx:2]: 🤪

**Review by wenche — APPROVED** [review]
That’s one small step for man, one giant leap for EDS. 👍 

Depending on what we do with the Storefront, we might have to update the examples there as well after release

File: libraries/core-react/src/components/Progress/Circular/CircularProgress.tokens.ts (hunk 1 above)
Comment (mimarz) [reply libraries/core-react/src/components/Progress/Circular/CircularProgress.tokens.ts:30]: done #1096

File: libraries/core-react/src/components/Progress/Dots/DotProgress.tsx (hunk 2 above)
Comment (mimarz) [reply libraries/core-react/src/components/Progress/Dots/DotProgress.tsx:2]: 🙈

**Review by pomfrida — APPROVED** [review]
LGTM 👍

**Pull request merged**